crates/libcontainer/src/process/container_main_process.rs

```diff
@@ -110,13 +96,22 @@ pub fn container_main_process(container_args: &ContainerArgs) -> Result<(Pid, Pi
 
     log::debug!("init pid is {:?}", init_pid);
 
-    // here we send both intermediate and init pid, because :
-    // init pid is required for writing it to pid_file (if) given by the high-level runtime
-    // intermediate pid is required in the case when we call exec, as we need to wait for the
-    // intermediate process to exit, which itself waits for child process (the exec process) to exit
-    // in order to get the proper exit code. We cannot simply wait for the init_pid , that is the actual container
-    // process, as it is not (direct) child of our process
-    Ok((intermediate_pid, init_pid))
+    // Before the main process returns, we want to make sure the intermediate
+    // process is exit and reaped. By this point, the intermediate process
+    // should already exited successfully. If intermediate process errors out,
+    // the `init_ready` will not be sent.
+    match waitpid(intermediate_pid, None)? {
```

Comment:
Does this affect the exec command?

Reply:
No. This is reaping the intermediate process, so `create`, `run`, `exec` all have to reap this process correctly, which is expected to always exit and the `youki main` process should wait for the intermediate process to exit and reap it.

Reply:
:ok: